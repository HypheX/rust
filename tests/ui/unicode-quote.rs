// Provide a useful error message when attempting to escape a non-ascii quotation mark.
// <https://github.com/rust-lang/rust/issues/128858>
fn main() {
    dbg!("Is this \“ a small sized quote or a big sized quote. \“ ");
    //~^ ERROR 4:20: 4:21: unknown character escape: `\u{201c}`
    //~^^ ERROR 4:65: 4:66: unknown character escape: `\u{201c}`
}
