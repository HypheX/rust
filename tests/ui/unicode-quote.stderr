error: unknown character escape: `\u{201c}`
  --> $DIR/unicode-quote.rs:4:20
   |
LL |     dbg!("Is this \“ a small sized quote or a big sized quote. \“ ");
   |                    ^ unknown character escape
   |
   = help: \u{201c} is not an ascii quote, but may look like one in some fonts; consider writing it in its escaped form for clarity.
help: if you meant to use a unicode quote; consider using its escaped form for clarity
   |
LL -     dbg!("Is this \“ a small sized quote or a big sized quote. \“ ");
LL +     dbg!("Is this \u{201c} a small sized quote or a big sized quote. \“ ");
   |

error: unknown character escape: `\u{201c}`
  --> $DIR/unicode-quote.rs:4:65
   |
LL |     dbg!("Is this \“ a small sized quote or a big sized quote. \“ ");
   |                                                                 ^ unknown character escape
   |
   = help: \u{201c} is not an ascii quote, but may look like one in some fonts; consider writing it in its escaped form for clarity.
help: if you meant to use a unicode quote; consider using its escaped form for clarity
   |
LL -     dbg!("Is this \“ a small sized quote or a big sized quote. \“ ");
LL +     dbg!("Is this \“ a small sized quote or a big sized quote. \u{201c} ");
   |

error: aborting due to 2 previous errors

